src/main/java/com/iexec/worker/compute/ComputeManagerService.java

```diff
@@ -74,8 +81,56 @@ public boolean downloadApp(TaskDescription taskDescription) {
         if (!isDockerType || taskDescription.getAppUri() == null) {
             return false;
         }
+
+        final long pullTimeout = computeImagePullTimeout(taskDescription);
         return dockerService.getClient(taskDescription.getAppUri())
-                .pullImage(taskDescription.getAppUri());
+                .pullImage(taskDescription.getAppUri(), Duration.of(pullTimeout, ChronoUnit.MINUTES));
+    }
+
+    private final Map<Long, Long> alreadyComputedTimeouts = new HashMap<>(5);
```

Comment:
Move it to file header section

Reply:
See https://github.com/iExecBlockchainComputing/iexec-worker/pull/442/commits/f164af0b2c3a0147402778cdd054d5b6cdf6ff46.